x/incentives/keeper/gauge.go

```diff
@@ -546,3 +546,59 @@ func (k Keeper) initGaugeInfo(ctx sdk.Context, poolIds []uint64) (types.Internal
 		GaugeRecords: gaugeRecords,
 	}, nil
 }
+
+// GetPoolIdsAndDurationsFromGroup retrieves the pool IDs and their associated durations from a given group.
+// It iterates over the gauge records in the group's InternalGaugeInfo, and for each record, it retrieves the pool ID and duration.
+// The function returns two slices: one for the pool IDs and one for the durations. The indices in these slices correspond to each other.
+// If there is an error retrieving the pool ID and duration for any gauge record, the function returns an error.
+func (k Keeper) GetPoolIdsAndDurationsFromGroup(ctx sdk.Context, group types.Group) ([]uint64, []time.Duration, error) {
```

Comment:
Do we need to provide the entire `types.Group`? I think `GaugeRecords` would contain all the necessary data needed to achieve the retrieval of pool IDs and durations. 

If agreement, the semantics would change from being "FromGroup" to being "FromGaugeRecords"

Reply:
changed here https://github.com/osmosis-labs/osmosis/pull/6601/commits/36ff2ea243bc4d4047ff8752678acb4da5df96dc